sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/models/Field.java

```diff
@@ -759,4 +771,35 @@ public Field setFields(List<Field> fields) {
         this.fields = fields;
         return this;
     }
+
+    /**
+     * Get the hidden property: A value indicating whether the field will be
+     * returned in a search result. This property must be false for key fields,
+     * and must be null for complex fields. You can hide a field from search
+     * results if you want to use it only as a filter, for sorting, or for
+     * scoring. This property can also be changed on existing fields and
+     * enabling it does not cause an increase in index storage requirements.
+     *
+     * @return the hidden value.
+     */
+    public Boolean isHidden() {
+        return retrievable == null ? null : !retrievable;
+    }
+
+    /**
+     * Set the hidden property: A value indicating whether the field will be
+     * returned in a search result. This property must be false for key fields,
+     * and must be null for complex fields. You can hide a field from search
+     * results if you want to use it only as a filter, for sorting, or for
+     * scoring. This property can also be changed on existing fields and
+     * enabling it does not cause an increase in index storage requirements.
+     *
+     * @param hidden the hidden value to set.
+     * @return the Field object itself.
+     */
+    public Field setHidden(Boolean hidden) {
+        retrievable = this.hidden == null ? null : !this.hidden;
```

Comment:
This would be easier to understand if you do:
```java
if (this.hidden != null) {
    this.retrievable = !this.hidden;
}
```
Shouldn't the retrievable logic be done after setting the member variable? not before?

Reply:
setHidden is the only entry to set `retrievable` field.
The logic needs to change to 
```
if (hidden != null) {
    this.retrievable = hidden;
}
this.hidden = hidden;
```
Good catch

Reply:
One line change is easy to do in transformer. :)

I would like to switch to 
```
this.hidden = hidden;
retrievable = this.hidden == null ? null : !this.hidden;
```